tests/framework/clients/rancher/client.go

```diff
@@ -88,13 +93,66 @@ func clientOpts(restConfig *rest.Config, rancherConfig *Config) *clientbase.Clie
 	}
 }
 
-// AsUser accepts a user object, and then creates a token for said `user`. Then it instantiates and returns a Client using the token created
-func (c *Client) AsUser(user *management.User) (*Client, error) {
-	token := &management.Token{
-		UserID: user.ID,
+func (c *Client) doAction(endpoint string, body []byte, into interface{}) error {
+	url := "https://" + c.restConfig.Host + endpoint
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
 	}
 
-	returnedToken, err := c.Management.Token.Create(token)
+	req.Header.Add("Authorization", "Bearer "+c.restConfig.BearerToken)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.Management.APIBaseClient.Ops.Client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		return httperror.NewAPIErrorLong(resp.StatusCode, resp.Status, url)
+	}
+
+	byteContent, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if len(byteContent) > 0 {
+		err = json.Unmarshal(byteContent, into)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	return fmt.Errorf("received empty response")
+}
+
+func (c *Client) getToken(user *management.User) (*management.Token, error) {
+	token := &management.Token{}
```

Comment:
I think we should move these functions outside of `client.go`. Although they are private, I don't think they should belong here. Unless you have a valid reason. I think the implementation is good, I'm just not convince they should be in this file, since it's so specific to getting a token from a created user. May they should go in extensions/users/users.go